src/app/clients/add/formComponents.tsx

```diff
@@ -0,0 +1,534 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import { useRouter } from "next/navigation";
+import {
+    setError,
+    setField,
+    Fields,
+    FormErrors,
+    getOnChange,
+    getNumberAdults,
+    numberRegex,
+    phoneNumberRegex,
+    postcodeRegex,
+    Error,
+    Person,
+    getChild,
+    onChangeCheckbox,
+    getBaby,
+    checkErrorOnSubmit,
+    insertClient,
+    insertFamily,
+    ClientDatabaseRecord,
+    formatPhoneNumber,
+    formatPostcode,
+    checkboxGroupToArray,
+    maxNumberChildren,
+    errorText,
+} from "@/app/clients/add/formFunctions";
+import FreeFormTextInput from "@/components/DataInput/FreeFormTextInput";
+import { errorExists } from "@/app/clients/add/formFunctions";
+import styled from "styled-components";
+import DropdownListInput from "@/components/DataInput/DropdownListInput";
+import CheckboxGroupInput from "@/components/DataInput/CheckboxGroupInput";
+import RadioGroupInput from "@/components/DataInput/RadioGroupInput";
+
+const CenterComponent = styled.div`
+    display: flex;
+    justify-content: center;
+    align-content: center;
+    margin: 2em;
+`;
+
+const StyledForm = styled.form`
+    width: 90%;
+    display: flex;
+    justify-content: center;
+    align-content: center;
+    flex-direction: column;
+`;
+
+const StyledCard = styled.div`
+    display: flex;
+    flex-direction: column;
+    padding: 2em;
+    margin: 2em;
+    width: 100%;
+    height: 80%;
+    border-radius: 10px;
+    background-color: ${(props) => props.theme.surfaceBackgroundColor};
+    color: ${(props) => props.theme.surfaceForegroundColor};
+
+    div {
+        width: 100%;
+        margin: 0.15em 0;
+    }
+`;
+
+const Text = styled.p`
+    padding-bottom: 1em;
+`;
+
+const Subheading = styled.h2`
+    padding-bottom: 1em;
+`;
+
+const Heading = styled.h1`
+    padding-bottom: 1em;
+`;
+
+const RequiredAsterisk = styled.span`
+    color: ${(props) => props.theme.errorColor};
+
+    &:before {
+        content: "*";
+    }
+`;
+
+const ErrorText = styled(Text)`
+    color: ${(props) => props.theme.errorColor};
+    margin-bottom: 3em;
+    text-align: center;
+`;
+
+const StyledButton = styled.button`
+    text-align: center;
+    width: 150px;
+    height: 40px;
+    border-radius: 10px;
+    border: solid 0 ${(props) => props.theme.foregroundColor};
+    background-color: ${(props) => props.theme.primaryBackgroundColor};
+    color: ${(props) => props.theme.primaryForegroundColor};
+
+    &:hover {
+        background-color: ${(props) => props.theme.secondaryBackgroundColor};
+        color: ${(props) => props.theme.secondaryForegroundColor};
+    }
+`;
+
+const genericFormCard = (
+    title: string,
+    required: boolean,
+    childComponent: React.ReactNode,
+    text?: string
+): React.ReactNode => {
```

Comment:
React.FC is better (where possible) than ReactNode, as well as being more specific it means you can interface the types out :)